src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,132 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_LENGTH = 10;
+    private T[] data;
+    private int size;
+
+    public ArrayList() {
+        data = (T[]) new Object[DEFAULT_LENGTH];
+        size = 0;
```

Comment:
I think it is redundant.
In java default int value is 0.
https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html